src/SmoFacade/Listener.cs

```diff
@@ -135,14 +135,33 @@ private static string AgListenerName(string dataSource)
     {
       var parts = agInstanceName.Split('\\');
       if(parts.Length == 1)
-        connBuilder.DataSource = Dns.GetHostEntry(agInstanceName).HostName;
+        connBuilder.DataSource = ResolveDnsAndGetHostName(connBuilder.DataSource, agInstanceName);
       else if(parts.Length == 2)
         // NamedInstances: chop instance name, resolve DNS, slap instance name back on!
-        connBuilder.DataSource = $"{Dns.GetHostEntry(parts[0]).HostName}\\{parts[1]}";
+        connBuilder.DataSource = $"{ResolveDnsAndGetHostName(connBuilder.DataSource, parts[0])}\\{parts[1]}";
       else
         throw new ArgumentException($"agInstanceName param {agInstanceName} cannot be resolved by DNS");
 
       return new Server(connBuilder.ToString(), credentialName);
     }
+
+    /// <summary>
+    ///   Resolves "instanceName" to a FQDNS name
+    ///   However on Unix OS, "instanceName" should be a complete domain - "{sub domain}.{second level domain}.{...}.{top level domain}" (eg: "xyz.abc.def.com")
+    ///   Therefore, append from the {second-level} -> {top-level} domains from the "hostDomain" to the "instanceName" to make it a complete domain if it is not already
+    /// </summary>
+    /// <param name="hostDomain">The complete domain for the SQL server</param>
+    /// <param name="instanceName">The specific name of an instance within the server</param>
+    private static string ResolveDnsAndGetHostName(string hostDomain, string instanceName)
+    {
+      if (Environment.OSVersion.Platform == PlatformID.Unix && !instanceName.Contains('.'))

```

Comment:
had to change 
from `RuntimeInformation.IsOSPlatform(OSPlatform.Linux)` 
to `Environment.OSVersion.Platform == PlatformID.Unix`
due to .Net Framework 4.7 compatibility issues